### PR TITLE
Add testing using kind


### PR DESCRIPTION
**Pull request by evrardjp**

### CODE REVIEW

File: kind.sh [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+EOF
+}
+
+check_all_nodes_are_ready() {
+	$kubectl get nodes | grep Ready | wc -l | grep $nodecount > /dev/null
+}
+
+spinup_cluster() {
+	gen_kind_manifest
+	kind create cluster --name "${KIND_CLUSTER_NAME}" --config=$tmp_dir/kind.yaml
```
Comment (dholbach) [inline kind.sh:120]: Maybe first check if `kind` is installed?

File: kind.sh [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+#KIND_IMAGE=kindest/node:v1.18.2
+#KIND_IMAGE=kindest/node:v1.18.0@sha256:0e20578828edd939d25eb98496a685c76c98d54084932f76069f886ec315d694
+#KIND_IMAGE=kindest/node:v1.17.0@sha256:9512edae126da271b66b990b6fff768fbb7cd786c7d39e86bdf55906352fdf62
+#KIND_IMAGE=kindest/node:v1.16.4@sha256:b91a2c2317a000f3a783489dfb755064177dbc3a0b2f4147d50f04825d016f55
+KIND_IMAGE=${KIND_IMAGE:-"kindest/node:v1.18.2"}
+
+# desired cluster name; default is "kured"
+KIND_CLUSTER_NAME="${KIND_CLUSTER_NAME:-kured}"
+
+KURED_IMAGE_DEST=${KURED_IMAGE_DEST:-"docker.io/evrardjp/kured:latest"}
```
Comment (dholbach) [inline kind.sh:30]: Maybe complain if `KURED_IMAGE_DEST` is not set instead?

**Comment by dholbach** [comment]
We could also take a look at [what Flux does](https://github.com/fluxcd/flux/tree/master/test/e2e) and steal some of their work. They use [bats](https://github.com/bats-core/bats-core) and kind for their e2e tests using GH Actions.

File: kind.sh (hunk 2 above)
Comment (evrardjp) [reply kind.sh:30]: Ok.

File: kind.sh (hunk 1 above)
Comment (evrardjp) [reply kind.sh:120]: ok.

**Comment by evrardjp** [comment]
I had a brief look into what Flux does this morning, and very little can be reused. I will have a look at bats in more details if it would bring us more flexibility into the testing. I can see it interesting now to test the installation of kured in different ways (manifests/helm chart), and do the same functional test (and improve when we'll get more functional tests).

**Comment by evrardjp** [comment]
I am not sure introducing a dependency with bats is necessary. I am refactoring this code for re-use in any case, to have a library of shell actions. I found a reliability pain point in this code, which I am now targetting.

**Comment by davidkarlsen** [comment]
Could https://github.com/marketplace/actions/kind-kubernetes-in-docker-action (or other action alternatives) be of use?

**Comment by dholbach** [comment]
Yep, I would think so - it's what Flux uses too.

**Comment by evrardjp** [comment]
I checked that action before my long holiday :)
I eventually found out that it was less flexible for my test cases.

Because one does not prevent another, I will propose a different approach on this patch, by ensuring we have a kind config in tree, and have manual/automated (github action) testing using it. This would be flexible enough for all the cases.

**Comment by evrardjp** [comment]
This is superseded by #183 .

**Pull request closed without merging**